src/ucp/dt/dt.h

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2016.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+
+#ifndef UCP_DT_H_
+#define UCP_DT_H_
+
+#include <ucp/dt/dt_contig.h>
+#include <ucp/dt/dt_iov.h>
+#include <ucp/dt/dt_generic.h>
+#include <ucp/core/ucp_request.h>
+
+
+/**
+ * Get the total length of the data
+ */
+static inline
+size_t ucp_dt_length(ucp_datatype_t datatype, size_t count,
+                     const ucp_dt_iov_t *iov, const ucp_frag_state_t *state)
+{
+    ucp_dt_generic_t *dt_gen;
+    size_t length = 0;
+
+    switch (datatype & UCP_DATATYPE_CLASS_MASK) {
+    case UCP_DATATYPE_CONTIG:
+        length = ucp_contig_dt_length(datatype, count);
```

Comment:
could be written simpler w/o extra variable "length" - just return the value directly from 'case' instead of doing "break"